### ISSUE TITLE
Do not remove duplicates if `$history_remove_dups` is unset

### ISSUE THREAD

**Issue by whitney-cumber** [neomutt/neomutt#3625]
## Expected Behaviour

Duplicates are recorded if `$history_remove_dups` is unset.

## Actual Behaviour

Some duplicates are removed even if `$history_remove_dups` is unset.

## Steps to Reproduce

0. Start Neomutt: `neomutt -F /dev/null`
1. Enter commands:
```
:echo 1
:echo 1
```
2. Enter command line and press "up"
3. Notice that only one copy of `echo 1` is recorded.

This can also be verified by tracking the contents of  `$history_file` if set.

## How often does this happen?

- Always

## When did it start to happen?

When I tested for it.

## NeoMutt Version

```
20220429
```

## Extra Info

Relevant code seems to be:

https://github.com/neomutt/neomutt/blob/c9f5b3f39ce98fb2d809763ed1697017920b888a/history/history.c#L499

One could argue that this is not a bug but intended behaviour.  Nevertheless, given that there is an extra config option to suppress duplicates, this behaviour is more than surprising so its worth thinking about its future (hence an issue and not a PR).

**Comment by gahr** [comment]
So we have two orthogonal features:
1. if you issue the same command multiple times in a row, we only record it once. This is not affected by any config.
2. whenever you issue a command, if `history_remove_dups` is set, the whole history is scanned and any precedent recording of the same command is removed. Then the command is added as the last issued command.

Perhaps it would be a good idea to add a mention of point 1. to the docs for `history_remove_dups`.

**Comment by whitney-cumber** [comment]
> So we have two orthogonal features:

I don't agree that they are orthogonal, simply because 2. ("always remove dups") implies 1. ("sometimes remove dups").

**Comment by gahr** [comment]
ok perhaps they are at some 30% :) I think it's reasonable to say "we never record subsequent identical commands more than once, and we never record identical commands more than once if you set a specific flag".

**Comment by Freed-Wu** [comment]
> we never record subsequent identical commands more than once

Record subsequent identical is meaningless except to provide the number of times this command has been executed. And it will increase the number of press Ctrl-P to search previous command. Many TUI programs do same behaviour, too.

> we should close this then

Done.

**Comment by whitney-cumber** [comment]
> Record subsequent identical is meaningless

I want to object to this description, my use case is the following.  I have some sequence of commands (for example to test something), let's say these are
```
foo
bar
bar
baz
```
Then I change some config or so (`set cool_config`), and then want to rerun these commands. To do this I can mechanically do prev-history 5 times[1] then press enter, and repeat that additional three times. No thinking or verifying commands necessary, it can even be automated.  This reproduces `foo`, `bar`, `bar`, `baz`, reliably.

However, this only works if `bar` is recorded twice.  Otherwise the number of repeats changes throughout the process:

* 4 for `foo`,
* 4 for `bar`,
* 5 (or 1) for `bar`,
* 4 for `baz`.

Meaning, I now have to keep track on where I am in the sequence of commands to react appropriately.

> Many TUI programs do same behaviour, too.

And many do not, e.g. bash, python3.

[1] 1 time for `set coll_config` and 4 times for `baz` to `foo`
